pkg/resource/deploy/snapshot.go

```diff
@@ -175,3 +178,45 @@ func (snap *Snapshot) VerifyIntegrity() error {
 
 	return nil
 }
+
+// Performs glob style expansion on urns that contain '*'. Each urn can be
+// expanded into 0-n actual urns, depending on what underlying resources exist
+// in the snapshot. URNs are returned in sorted order. All returned urns are unique.
+func (snap *Snapshot) GlobUrn(urn resource.URN) []resource.URN {
+	if !strings.Contains(string(urn), "*") {
+		return []resource.URN{urn}
```

Comment:
This case is bit weird. I think the weirdness comes from GlobUrn combining two distinct pieces of functionality - filtering `snap.Resources` and matching against a pattern. This clause returns `urn` even if it's not mentioned in `snap.Resources`. You would think that the return values `[]resource.URN` are always a subset so this may be a little surprising. I'm guessing this is exactly what you want at the call sites but it's not obvious in the func itself.

Reply:
It's also how a normal glob function works (for example, in sh). Expanding an argument with no wild cards returns that argument, even if the argument is not a valid path. 
```
$ echo *.txt
real_file.txt
$ echo fake.txt
fake.txt
```